server/services/participants.js

```diff
@@ -125,6 +125,52 @@ const updateParticipant = async (participantInfo) => {
   return participant;
 };
 
+const validateConfirmationId = (id) =>
+  dbClient.db[collections.CONFIRM_INTEREST].findOne({ otp: id });
+
+const confirmParticipantInterest = async (id) => {
+  const now = new Date().toJSON();
+
+  const relatedParticipants = await dbClient.db[collections.PARTICIPANTS]
+    .join({
+      [collections.CONFIRM_INTEREST]: {
+        type: 'INNER',
+        on: { email_address: 'body.emailAddress', otp: id },
+      },
+    })
+    .find();
+
+  const updatedParticipantFields = relatedParticipants.map((participant) => ({
+    id: participant.id,
+    userUpdatedAt: now,
+    interested: 'yes',
+    history: [
+      {
+        changes: [
+          {
+            to: now,
+            from: participant.body.userUpdatedAt,
+            field: 'userUpdatedAt',
+          },
+        ],
+        timestamp: now,
+      },
+      ...(participant.body.history ? participant.body.history : []),
+    ],
+  }));
+
+  await Promise.all(
+    updatedParticipantFields.map((fields) => {
+      const result = dbClient.db[collections.PARTICIPANTS].updateDoc({ id: fields.id }, fields);
```

Comment:
Ah ha! This outta do it.
```suggestion
    updatedParticipantFields.map(({id, ...fields}) => {
      const result = dbClient.db[collections.PARTICIPANTS].updateDoc({ id }, fields);
```

Reply:
Ah, so the `fields` object contained an `id` that was updating the pk of the entry, and {id, ...fields} expands the object to pull out the id as a separate value from the rest of `fields`?